docs/reference/eql/functions.asciidoc

```diff
@@ -805,6 +806,98 @@ If using a field as the argument, this parameter supports only
 *Returns:* integer, float, or `null`
 ====
 
+[discrete]
+[[eql-fn-number]]
+=== `number`
+
+Converts a string to the corresponding integer or float.
+
+[%collapsible]
+====
+*Example*
+[source,eql]
+----
+number("1337")              // returns 1337
+number("42.5")              // returns 42.5
+number("deadbeef", 16)      // returns 3735928559
+
+// integer literals beginning with "0x" are auto-detected as hexadecimal
+number("0xdeadbeef")        // returns 3735928559
+number("0xdeadbeef", 16)    // returns 3735928559
+
+// "+" and "-" are supported
+number("+1337")             // returns 1337
+number("-1337")             // returns -1337
+
+// surrounding whitespace is ignored
+number("  1337  ")          // returns 1337
+
+// process.pid = "1337"
+number(process.pid)         // returns 1337
+
+// null handling
+number(null)                // returns null
+number(null, 16)            // returns null
+number("1337", null)        // returns null
```

Comment:
This should actually return `1337`
Unfortunately, there's not a good way for us to tell if an argument evaluated  to `null`, if `null` was passed in, or if the argument wasn't specified at all. This should technically return `1337`

Reply:
:doh: thanks for catching this. I thought I had tested this one, but it must have slipped by. Corrected with 626eefe10272d3cdd62525aea9acbf378b1a806b.